### PR TITLE
Syntax changes to support 'scoped'

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

File: proposals/low-level-struct-improvements.md [hunk 1]
```diff
@@ -410,6 +410,86 @@ Detailed Notes:
     - The core library contains the feature flag indicating support for `ref` fields
     - The `langversion` value is 11 or higher
 
+### Syntax
+[12.6.2 Local variable declarations](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/statements.md#1262-local-variable-declarations): added `local_variable_scope_modifier`.
+```antlr
+local_variable_declaration
+    : local_variable_scope_modifier? local_variable_type local_variable_declarators
```
Comment (jcouv) [inline proposals/low-level-struct-improvements.md:417]: nit: `local_variable_scope_modifier` -> `local_variable_type_modifier`?

File: proposals/low-level-struct-improvements.md [hunk 2]
```diff
@@ -410,6 +410,86 @@ Detailed Notes:
...
+fixed_parameter
+    : attributes? parameter_modifier? type identifier default_argument?
+    ;
+
+parameter_modifier
+    : parameter_scope_modifier
+    | 'this' parameter_scope_modifier?
+    ;
+
+parameter_scope_modifier
```
Comment (jcouv) [inline proposals/low-level-struct-improvements.md:461]: Similar question here. I think the organization makes sense, but I'm not sure about the name.

File: proposals/low-level-struct-improvements.md [hunk 3]
```diff
@@ -410,6 +410,86 @@ Detailed Notes:
...
+    : anonymous_function_parameter_modifier
+    | 'scoped' anonymous_function_parameter_modifier?
+    ;
+
+anonymous_function_parameter_modifier
+    : 'in'
+    | 'ref'
+    | 'out'
+    ;
+```
```
Comment (jcouv) [inline proposals/low-level-struct-improvements.md:491]: We'll also need to update `for` statements and declaration expressions (discussed in https://github.com/dotnet/roslyn/issues/62039).

Comment (cston) [reply proposals/low-level-struct-improvements.md:491]: `for` statements and declaration expressions are described above (see references to sections 12.9.4 and 11.6.2).

Comment (jcouv) [reply proposals/low-level-struct-improvements.md:477]: Section 11.6.2 is for arguments, so handles `scoped out var x`. But it doesn't cover declaration expressions such as `(scoped var x, scoped var y) = M();`.

**Comment by AlekseyTs** [comment]
@cston Is this PR work in progress? O can we merge it?

**Comment by cston** [comment]
@dotnet/roslyn-compiler, please review, thanks.

File: proposals/low-level-struct-improvements.md [hunk 4]
```diff
@@ -410,6 +410,77 @@ Detailed Notes:
...
+    : expression
+    | 'in' variable_reference
+    | 'ref' variable_reference
+    | 'scoped'? 'out' local_variable_type? identifier
+    ;
+```
+
+[--.-.- Deconstruction expressions](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md):
+```antlr
+[TBD]
```
Comment (333fred) [inline proposals/low-level-struct-improvements.md:447]: Did you want to address this before merge?

Comment (cston) [reply proposals/low-level-struct-improvements.md:447]: I'd rather handle this in a separate PR. Deconstruction expressions seem distinct from the other cases here, and the this syntax may depend on whether we have a common form for 'out' variables and deconstruction expressions in general in the spec.

**Review by 333fred — APPROVED** [review]

**Pull request merged**